lib/protoboeuf/codegen.rb

```diff
@@ -2,10 +2,12 @@
 
 require "erb"
 require "syntax_tree"
-require_relative "codegen_type_helper"
 
 module ProtoBoeuf
   class CodeGen
+    require "protoboeuf/codegen/type_helper"
+    require "protoboeuf/codegen/field"
```

Comment:
Not 100% sure whether `require` inside the class is more appropriate than `autoload`.  See this discussion: 

https://github.com/Shopify/protoboeuf/pull/183#pullrequestreview-2584637485
https://github.com/Shopify/protoboeuf/pull/183#issuecomment-2625403381
https://github.com/Shopify/protoboeuf/pull/183#issuecomment-2628035354

Reply:
I'm also not convinced this is worth blocking this PR over, but I'd like to resolve this as a best practice in the near future.